tests/test_adjoint.py

```diff
@@ -59,6 +63,43 @@ def test_adjoint_F(self, mkey, shape, kernel, space_order):
              % (term1, term2, (term1 - term2)/term1, term1 / term2))
         assert np.isclose((term1 - term2)/term1, 0., atol=1.e-12)
 
+    @pytest.mark.parametrize('mkey, shape, space_order', [
+        # 2D tests with varying space orders
+        ('layers', (30, 35), 8), ('layers', (30, 35), 4),
+        # 3D tests with varying space orders
+        ('layers', (30, 35, 40), 8), ('layers', (30, 35, 40), 4),
+    ])
+    def test_adjoint_tti_F(self, mkey, shape, space_order):
```

Comment:
This can probably be merged with above test with 
`test_adjoint_f(self, mkey, shape, space_order, setup_func)`
and
solver = `setup_func` ....